docs/guides/moving-data.md

```diff
@@ -0,0 +1,325 @@
+
+=== "AWS"
+
+    [prefect-aws](https://prefecthq.github.io/prefect-aws/) provides blocks for interacting with AWS services.
+
+    ```bash
+    pip install -U prefect-aws
+    ```
+
+=== "Azure"
+
+    [prefect-azure](https://prefecthq.github.io/prefect-azure/) provides blocks for interacting with Azure services.
+
+    ```bash
+     pip install -U prefect-azure
+    ```
+
+=== "GCP"
+
+    [prefect-gcp](https://prefecthq.github.io/prefect-gcp/) provides blocks for interacting with GCP services.
+    
+    ```bash
+     pip install -U prefect-gcp
+    ```
+
+## Register the block types
+
+Register the new block types with Prefect Cloud (or with your self-hosted Prefect server instance):
+
+=== "AWS"
+
+    ```bash
+    prefect block register -m prefect_aws  
+    ```
+
+=== "Azure"
+
+    ```bash
+    prefect block register -m prefect_azure 
+    ```
+
+=== "GCP"
+
+    ```bash
+    prefect block register -m prefect_gcp
+    ```
+
+We should see a message in the CLI that several block types were registered.
+If we check the UI, we should see the new block types listed.
+
+## Create a storage bucket
+
+Create a storage bucket in the cloud provider account.
+Ensure the bucket is publicly accessible or create a user or service account with the appropriate permissions to fetch and write data to the bucket.
+
+## Create a credentials block
+
+If the bucket is private, there are several options to authenticate:
+
+1. Use a worker that is authenticated at deployment run time.
+2. Create a block with configuration details and reference it when creating the storage block.
+
+If saving credential details in a block we can use a credentials block specific to the cloud provider or use a more generic secret block.
+We can create [blocks](/concepts/blocks/) via the UI or Python code.
+Below we'll use Python code to create a credentials block for our cloud provider.
+
+!!! Warning "Credentials safety"
+
+    Reminder, don't store credential values in public locations such as public git platform repositories. 
+    In the examples below we use environment variables to store credential values.
+
+=== "AWS"
+
+    ```python
+    import os
+    from prefect_aws import AwsCredentials
+
+    my_aws_creds = AwsCredentials(
+        aws_access_key_id="123abc",
+        aws_secret_access_key=os.environ.get("MY_AWS_SECRET_ACCESS_KEY"),
+    )
+    my_aws_creds.save(name="my-aws-creds-block", overwrite=True)
+    ```
+
+=== "Azure"
+
+    ```python
+    import os
+    from prefect_azure import AzureBlobStorageCredentials
+
+    my_azure_creds = AzureBlobStorageCredentials(
+        connection_string=os.environ.get("MY_AZURE_CONNECTION_STRING"),
+    )
+    my_azure_creds.save(name="my-azure-creds-block", overwrite=True)
+    ```
+
+=== "GCP"
+
+    We recommend using specifying the service account key file contents as a string, rather than the path to the file, because that file will not be available in most production environments.
```

Comment:
```suggestion
    We recommend specifying the service account key file contents as a string, rather than the path to the file, because that file might not be available in your production environments.
```